### PR TITLE
chore(deps): upgrade all the AWS SDK v2 deps, and s3iofs

### PR DESCRIPTION
**Pull request by wolfeidau**
This resolves issues with endpoint changes in the SDK, see https://github.com/aws/aws-sdk-go-v2/issues/2370, and upgrades s3iofs which has been upgraded to improve performance, see https://github.com/wolfeidau/s3iofs/pull/25.

I have tested this in a project I started at https://github.com/wolfeidau/iceberg-go-cli cc @zeroshade

### CODE REVIEW

**Review by nastra — APPROVED** [review]
LGTM, will wait for @zeroshade to review this before merging

**Review by zeroshade — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
